Max-Sv/-aws-shop-be
# Task 7

Task: https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/7_authorization/task.md
Link to FE deploy: https://d2sqr7ze13s3vc.cloudfront.net/
Link to FE PR : https://github.com/Max-Sv/shop-angular-cloudfront/pull/3

Credentials to check:
Login: MaxSv
Password: TEST_PASSWORD
Auth token: TWF4U3Y6VEVTVF9QQVNTV09SRA==
LocalStorage params: authorization_token  TWF4U3Y6VEVTVF9QQVNTV09SRA==
csv file example:
count,description,price,title
2,test1,33,test1
3,test2,23,test2

Score: 6/6

Done:

- authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file

-  Import Service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.

-  update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser [localStorage](https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage) const authorization_token = localStorage.getItem('authorization_token')

Additional (optional) tasks

- Client application should display alerts for the responses in 401 and 403 HTTP statuses.
